device/iothub_client_python/src/iothub_client_python.cpp

```diff
@@ -1513,6 +1514,17 @@ class IoTHubClient
         {
             throw IoTHubClientError(__func__, IOTHUB_CLIENT_ERROR);
         }
+        else
+        {
+            if (client_interface_type == CLIENT_INTERFACE_DEVICE)
```

Comment:
>if (client_interface_type == CLIENT_INTERFACE_DEVICE) [](start = 12, length = 53)

Nit - but I'd recommend having a common function like IoTHubClient::SetProductInfo() that all the other CreateHandle's call into.  This way if there ever turns out to be other ways of having a CLIENT_INTERFACE_MODULE, you don't sprinkle this logic all over the place but instead just have it work from get-go.